ska_dbi/sqsh.py

```diff
@@ -0,0 +1,93 @@
+import subprocess
+from pathlib import Path
+from astropy.table import Table
+
+SYBASE = "/soft/SYBASE16.0"
+ENV_VARS = [
+    f"SYBASE={SYBASE}",
+    "LD_LIBRARY_PATH=/soft/SYBASE16.0/OCS-16_0/lib:/soft/SYBASE16.0/OCS-16_0/lib3p64:/soft/SYBASE16.0/OCS-16_0/lib3p",
+]
+SQSH_BIN = "/usr/local/bin/sqsh.bin"
+
+
+class Sqsh(object):
+    def __init__(
+        self,
+        server=None,
+        user=None,
+        passwd=None,
+        database=None,
+        sqshrc=None,
+        authdir="/proj/sot/ska/data/aspect_authorization",
+        **kwargs,
+    ):
+        DEFAULTS = {"server": "sqlsao", "user": "aca_ops", "database": "axafapstat"}
+
+        self.server = server or DEFAULTS.get("server")
+        self.user = user or DEFAULTS.get("user")
+        self.database = database or DEFAULTS.get("database")
+        self.passwd = passwd
+        self.sqshrc = sqshrc
+
+        if not Path(SYBASE).exists():
+            raise RuntimeError("SYBASE does not exist: %s" % SYBASE)
+        if not Path(SQSH_BIN).exists():
+            raise RuntimeError("sqsh.bin does not exist: %s" % SQSH_BIN)
+
+        if self.passwd is None and self.sqshrc is None:
+            sqshrc_file = (
+                Path(authdir) / f"sqsh-{self.server}-{self.database}-{self.user}"
+            )
+            if sqshrc_file.exists():
+                self.sqshrc = sqshrc_file
+            else:
+                raise NoPasswordError(
+                    "None supplied and unable to read inferred sqshrc file %s" % e
+                )
+
+    def __enter__(self):
+        """Context manager enter runtime context.  No action required, just return self."""
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """
+        Context manager exit run time context.
+
+        Close connection.  By the implicit "return None" this will raise any exceptions
+        after closing.
+        """
+        pass
```

Comment:
As a plain cmd-line call, the Sqsh stuff doesn't really have a state that needs to be cleaned up to need a context manager.  However, I figured it was easier to "plug it in" to the previous calls to with ska_dbi.DBI if I just gave Sqsh `__enter__` and `__exit__`.  I'm not sure what the appropriate no-op for `__exit__` is though.